next/frontend/utils/amplify.ts

```diff
@@ -0,0 +1,244 @@
+import { APPROVED_SSO_ORIGINS } from './sso'
+import { verifyIdentityApi } from 'frontend/api/api'
+import { useRouter } from 'next/router'
+import { useEffect, useState } from 'react'
+
+export enum PostMessageTypes {
+  ACCESS_TOKEN = 'ACCESS_TOKEN',
+  UNAUTHORIZED = 'UNAUTHORIZED',
+}
+
+export interface CityAccountPostMessage {
+  type: PostMessageTypes
+  payload?: Record<string, string>
+}
+
+export interface AccountError {
+  message: string
+  code: string
+}
+
+export enum AccountStatus {
+  Idle,
+  NewPasswordRequired,
+  NewPasswordSuccess,
+  EmailVerificationRequired,
+  EmailVerificationSuccess,
+  IdentityVerificationRequired,
+  IdentityVerificationPending,
+  IdentityVerificationFailed,
+  IdentityVerificationSuccess,
+}
+
+export enum Tier {
+  NEW = 'NEW',
+  QUEUE_IDENTITY_CARD = 'QUEUE_IDENTITY_CARD',
+  NOT_VERIFIED_IDENTITY_CARD = 'NOT_VERIFIED_IDENTITY_CARD',
+  IDENTITY_CARD = 'IDENTITY_CARD',
+  EID = 'EID',
+}
+
+export interface Address {
+  formatted?: string
+  street_address?: string
+  locality?: string
+  region?: string
+  postal_code?: string
+  country?: string
+  phone_number?: string
+}
+
+export type AccountType = 'fo' | 'po'
+
+// as returned by Amplify.Auth.currentAuthenticatedUser().attributes
+// sent from BE as server side props
+export interface UserData {
+  sub?: string
+  email_verified?: string
+  email?: string
+  name?: string
+  given_name?: string
+  family_name?: string
+  phone_number?: string
+  phone_verified?: string
+  address?: string
+  'custom:ifo'?: string
+  'custom:tier'?: Tier
+  'custom:account_type'?: AccountType
+  'custom:turnstile_token'?: string
+}
+
+// non standard, has prefix custom: in cognito
+const customAttributes = new Set(['ifo', 'tier', 'account_type'])
+const updatableAttributes = new Set([
+  'name',
+  'given_name',
+  'family_name',
+  'phone_number',
+  'address',
+  'tier',
+  'account_type',
+])
+
+Amplify.configure({
+  Auth: {
+    // REQUIRED only for Federated Authentication - Amazon Cognito Identity Pool ID
+    identityPoolId: process.env.NEXT_PUBLIC_COGNITO_IDENTITY_POOL_ID,
+
+    // REQUIRED - Amazon Cognito Region
+    region: process.env.NEXT_PUBLIC_AWS_REGION,
+
+    // OPTIONAL - Amazon Cognito User Pool ID
+    userPoolId: process.env.NEXT_PUBLIC_COGNITO_USER_POOL_ID,
+
+    // OPTIONAL - Amazon Cognito Web Client ID (26-char alphanumeric string)
+    userPoolWebClientId: process.env.NEXT_PUBLIC_COGNITO_CLIENT_ID,
+
+    // OPTIONAL - Enforce user authentication prior to accessing AWS resources or not
+    mandatorySignIn: false,
+
+    // OPTIONAL - This is used when autoSignIn is enabled for Auth.signUp
+    // 'code' is used for Auth.confirmSignUp, 'link' is used for email link verification
+    signUpVerificationMethod: 'code',
+  },
+  ssr: true,
+})
+
+// TODO - could be better, currently used only after login, AccountStatus should be replaced or rewritten
+export const mapTierToStatus = (tier?: Tier): AccountStatus => {
```

Comment:
This is more typesafe
```
const tierStatusMap = {
  [Tier.QUEUE_IDENTITY_CARD]: AccountStatus.IdentityVerificationPending,
  [Tier.NOT_VERIFIED_IDENTITY_CARD]: AccountStatus.IdentityVerificationFailed,
  [Tier.IDENTITY_CARD]: AccountStatus.IdentityVerificationSuccess,
  [Tier.EID]: AccountStatus.IdentityVerificationSuccess,
  [Tier.NEW]: AccountStatus.Idle,
};

export const mapTierToStatus = (tier?: Tier): AccountStatus => {  
  return tierStatusMap[tier] || AccountStatus.IdentityVerificationRequired;
}

Reply:
planning to get rid of this completely - you can see first part (not using status and using tier directly) already in, still need to handle the stepper statuses in registration/activation/login, will remove afterwards